### PR TITLE
I5-4928: Per tenent logs

### PR DESCRIPTION
**Pull request by derek-oneill**

### CODE REVIEW

**Comment by derek-oneill** [comment]
paired with: https://github.com/entrinsik-org/i5/pull/3752

File: lib/index.js [hunk 1]
```diff
@@ -13,9 +13,12 @@ class RedisLogStream extends stream.Writable {
     }
 
     _write (chunk, enc, done) {
+        const message = JSON.parse(chunk);
+        const tenant = _.get(message, ['config', 'tenant'], 'manager');
+        const name = `${this.name}:${tenant}`
```
Comment (bleupen) [inline lib/index.js]: can you make this tiny change for me?

in the constructor, do this:

```javascript
this.name = _.isFunction(name) ? name : _.constant(name);
```

and then here change to:

```javascript
const name = this.name(message);
```

then we would configure the logger with a name() callback function that applies the tenant

File: lib/index.js [hunk 2]
```diff
@@ -13,9 +13,12 @@ class RedisLogStream extends stream.Writable {
     }
 
     _write (chunk, enc, done) {
+        const message = JSON.parse(chunk);
```
Comment (bleupen) [inline lib/index.js]: i hate having to serialize and deserialize for every log message. i wonder if we can ditch SaferToJson and just do the serialization here?

File: lib/index.js [hunk 3]
```diff
@@ -71,6 +74,11 @@ exports.register = function (server, opts, next) {
             _.set(req.plugins, ['good', 'requestResponseFilter'], _.get(req.server, ['app', ...settingsPath]));
             reply.continue();
         });
+
+        server.ext('onRequest', (req, reply) => {
```
Comment (bleupen) [inline lib/index.js]: i wonder if this shouldnt be pulled into the informer/server/log file. it feels weird to make this library aware of our multitenant implementation

**Pull request merged**